core/Segment.php

```diff
@@ -705,4 +705,15 @@ public function getOriginalString()
     {
         return $this->originalString;
     }
+
+    /**
+     * Escapes segment expression delimiters in a segment value with a backslash if not already done.
+     */
+    private function escapeSegmentValue(string $value): string
+    {
+        $delimiterPattern = SegmentExpression::AND_DELIMITER . SegmentExpression::OR_DELIMITER;
+        $pattern = '/((?<!\\\)[' . $delimiterPattern . '])/';
```

Comment:
Do we need `preg_quote`? I know we know what the chars are (now)... Just a mention, no need to action.

```suggestion
        $pattern = '/((?<!\\\)[' . preg_quote($delimiterPattern) . '])/';
```